src/pages/Result.tsx

```diff
@@ -9,27 +9,31 @@ import useFetch, { cityGroup } from '../hooks/useFetch';
 
 import { dustDataType } from '@/type';
 
-import axios from 'axios';
-const { VITE_API_KEY, VITE_OPEN_URL } = import.meta.env;
-
 const Result = () => {
-  // const data = useFetch();
   const [dustData, setDustData] = useState<dustDataType | []>([]);
   const location = useLocation();
   const choiceCity = location.state;
 
-  // data.then((data) => setDustData(data));
-  useEffect(() => {
-    Promise.all(
-      cityGroup.map((v) =>
-        axios
-          .get(
-            `${VITE_OPEN_URL}?sidoName=${v.cityName}&pageNo=1&numOfRows=100&returnType=json&serviceKey=${VITE_API_KEY}&ver=1.0`
-          )
-          .then((res) => res.data.response.body)
-      )
-    ).then((data) => setDustData(data));
-  }, [location]);
+  const data = useFetch();
+  data.then((data) => setDustData(data));
+
+  const findChoiceCity = (kindOfdust: string) => {
+    const result = dustData.find(
+      (temp) => temp.items[0].sidoName === choiceCity
+    );
+
+    if (result && kindOfdust === 'DustState') {
+      return (
+        (parseInt(result?.items[4]?.pm10Grade) +
+          parseInt(result?.items[4]?.pm25Grade)) /
+        2
+      ).toString();
+    } else if (result && kindOfdust === 'first') {
+      return result?.items[4]?.pm10Value;
+    } else if (result && kindOfdust === 'last') {
+      return result?.items[4]?.pm25Value;
+    } else return '';
+  };
```

Comment:
```suggestion
    if(result) return;
    
    if (kindOfdust === 'DustState') {
      return (
        (parseInt(result?.items[4]?.pm10Grade) +
          parseInt(result?.items[4]?.pm25Grade)) /
        2
      ).toString();
    } else if (kindOfdust === 'first') {
      return result?.items[4]?.pm10Value;
    } else if (kindOfdust === 'last') {
      return result?.items[4]?.pm25Value;
    } else return '';
  };
```

이런식은 어떤가요? 더 나아가서

```suggestion
    if(result) return;
    
    switch (kindOfDust) {
      case 'DustState':
        return (
          (parseInt(result.items[4].pm10Grade) + parseInt(result.items[4].pm25Grade)) / 2
        ).toString();
      case 'first':
        return result.items[4].pm10Value;
      case 'last':
        return result.items[4].pm25Value;
      default:
        return '';
    }
```

이렇게! 그리고 `first`, `last`보다 미세먼지, 초미세먼지임을 나타내주는 변수명이 더 좋을 것 같아요. 마지막으로 default로  `return ''`이 아닌 미세먼지나 초미세먼지 등을 default로 `return`해주는 게 더 바람직할 것 같기도해요.


정말정말 마지막으로 `findChoiceCity`이 함수명인데 이 함수 안에서 City도 찾고 미세먼지까지 찾아주는데 더 관심사를 분리할 수 있을 것 같아요. 먼가 짧은 코드 줄에서 5개정도 리뷰를 남겼는데 재관님이라면 멋지게 반영해주실 것 같아요!

Reply:
흠... 함수를 분리하긴 했는데 제가 원하는 대로 분리되진 않았어요.
findChoiceCity 함수와 calculateFineDust 함수로 분리 되어있긴 하지만 결국엔 findChoiceCity 함수에서 calculateFineDust 함수를 불러서 사용하기 때문에 깔끔하게 분리되어 있지 않아요.
calculateFineDust 함수의 의존성을 제거하려는 과정에서 한가지 막힘이 있었어요. findChoiceCity에서 사용하는 dustData.find의 결과값 result를 전역으로 빼주고 calculateFineDust에서 result를 가져다가 쓰려고 했어요.
그러나 이렇게 할 경우 상단에 내가 선택한 도시의 미세먼지 progressBar가 정상적으로 출력되지 않더라구요. 
미세먼지에 대한 정보를 state로 빼서 result에 디팬던시를 둔 useEffect를 활용해 볼까 했지만 이 역시 정상적으로 출력도지 않았어요.

혹시 좋은 아이디어가 있을까요?

Reply:
default로 return 해주는 value 의 경우 숫자가 들어가게 되므로 '0' 이라는 값을 기본값으로 생각했는데 더 적절한 값이 있을지 궁금해요.

Reply:
지금방법도 나쁘지않다고 생각해요. 이전보다 훨씬 좋아졌어요 ㅎㅎ 그리고 0도 지금은 괜찮은 것 같긴해요